src/components/PositionedOverlay/PositionedOverlay.tsx

```diff
@@ -238,17 +231,14 @@ export class PositionedOverlay extends React.PureComponent<
           activatorRect,
           overlayRect,
           containerRect,
-          overlayMargins,
           preferredAlignment,
         );
 
         this.setState(
           {
             measuring: false,
             activatorRect: getRectForNode(activator),
-            right:
-              preferredAlignment === 'right' ? horizontalPosition : undefined,
-            left: preferredAlignment === 'right' ? 0 : horizontalPosition,
+            left: horizontalPosition,
```

Comment:
Remove the right value and only positioning with left.

This simplifies the math when the popover overlaps the page. It now doesn't have to calculate a three right values it can just switch to a previously defined left value.